official/transformer/transformer_main.py

```diff
@@ -249,141 +265,185 @@ def train_schedule(
+      help=flags_core.help_wrap("The number of steps used to train."))
+  flags.DEFINE_integer(
+      name="steps_between_evals", short_name="sbe", default=1000,
+      help=flags_core.help_wrap(
+          "The Number of training steps to run between evaluations. This is "
+          "used if --train_steps is defined."))
 
-  # Determine training schedule based on flags.
-  if FLAGS.train_steps is not None and FLAGS.train_epochs is not None:
+  # BLEU score computation
+  flags.DEFINE_string(
+      name="bleu_source", short_name="bls", default=None,
+      help=flags_core.help_wrap(
+          "Path to source file containing text translate when calculating the "
+          "official BLEU score. --bleu_source, --bleu_ref, and --vocab_file "
+          "must be set. Use the flag --stop_threshold to stop the script based "
+          "on the uncased BLEU score."))
+  flags.DEFINE_string(
+      name="bleu_ref", short_name="blr", default=None,
+      help=flags_core.help_wrap(
+          "Path to source file containing text translate when calculating the "
+          "official BLEU score. --bleu_source, --bleu_ref, and --vocab_file "
+          "must be set. Use the flag --stop_threshold to stop the script based "
+          "on the uncased BLEU score."))
+  flags.DEFINE_string(
+      name="vocab_file", short_name="vf", default=VOCAB_FILE,
+      help=flags_core.help_wrap(
+          "Name of vocabulary file containing subtokens for subtokenizing the "
+          "bleu_source file. This file is expected to be in the directory "
+          "defined by --data_dir."))
+
+  flags_core.set_defaults(data_dir="/tmp/translate_ende",
+                          model_dir="/tmp/transformer_model",
+                          batch_size=None,
+                          train_epochs=None)
+
+
+def validate_flags(flags_obj):
+  """Raise ValueError if flags have invalid values."""
+  if flags_obj.train_steps is not None and flags_obj.train_epochs is not None:
     raise ValueError("Both --train_steps and --train_epochs were set. Only one "
                      "may be defined.")
-  if FLAGS.train_steps is not None:
-    train_eval_iterations = FLAGS.train_steps // FLAGS.steps_between_eval
-    single_iteration_train_steps = FLAGS.steps_between_eval
+
+  # Make sure that the BLEU source and ref files if set
+  if flags_obj.bleu_source is not None and flags_obj.bleu_ref is not None:
+    if not tf.gfile.Exists(flags_obj.bleu_source):
+      raise ValueError(
+          "BLEU source file %s does not exist" % flags_obj.bleu_source)
+    if not tf.gfile.Exists(flags_obj.bleu_ref):
+      raise ValueError(
+          "BLEU source file %s does not exist" % flags_obj.bleu_ref)
+    # Make sure that vocab file is in the data directory defined
+    vocab_file_path = os.path.join(flags_obj.data_dir, flags_obj.vocab_file)
+    if not tf.gfile.Exists(vocab_file_path):
+      raise ValueError("Vocabulary file %s does not exist" % vocab_file_path)
+
+
+def main(_):
```

Comment:
I think you can follow taylor's change in imagenet, which creates a wrapper of main function, so that flag object can be injected

Reply:
Indeed.